backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -482,3 +498,32 @@ def generate_default_kms_decrypt_policy_statement(target_requester_arn):
             "Action": "kms:Decrypt",
             "Resource": "*"
         }
+
+    @staticmethod
+    def generate_enable_pivot_role_permissions_policy_statement(source_account_id):
+        return {
+            "Sid": f"{DATAALL_BUCKET_ENABLE_PIVOT_ROLE_PERMISSIONS_SID}",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": [
+                    f"arn:aws:iam::{source_account_id}:role/dataallPivotRole"
+                ]
+            },
+            "Action": [
+                "kms:Decrypt",
+                "kms:Encrypt",
+                "kms:GenerateDataKey*",
+                "kms:PutKeyPolicy",
+                "kms:GetKeyPolicy",
+                "kms:ReEncrypt*",
+                "kms:TagResource",
+                "kms:UntagResource",
+                "kms:DescribeKey"
+               ],
+            "Resource": "*",
+            "Condition": {
```

Comment:
Do not think Condition needed in Key Policy

Reply:
Removed